src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs

```diff
@@ -310,7 +289,7 @@ public string EvaluateDebuggerDisplayString(string formatString)
                 throw new ArgumentException(string.Format("Unmatched open brace in '{0}'", formatString));
             }
 
-            return pooled.ToStringAndFree();
+            completionRoutine(new DkmEvaluateDebuggerDisplayStringAsyncResult(pooled.ToStringAndFree()));
```

Comment:
Consider completing this asynchronously, so we're testing the async case.
ThreadPool.QueueUserWorkItem(() => completionRoutine(...));